services/galley/test/integration/API.hs

```diff
@@ -888,6 +900,37 @@ testAddRemoteMember = do
     let expected = [OtherMember remoteBob Nothing roleNameWireAdmin]
     assertEqual "other members should include remoteBob" expected actual
 
+testAddRemoteMemberFailure :: TestM ()
+testAddRemoteMemberFailure = do
+  alice <- randomUser
+  bobId <- randomId
+  charlieId <- randomId
+  let remoteDomain = Domain "far-away.example.com"
+      remoteBob = Qualified bobId remoteDomain
+      remoteCharlie = Qualified charlieId remoteDomain
+  convId <- decodeConvId <$> postConv alice [] (Just "remote gossip") [] Nothing Nothing
+  opts <- view tsGConf
+  g <- view tsGalley
+  liftIO $ do
+    (resp, _) <-
+      withTempMockFederator
+        opts
+        remoteDomain
+        [mkProfile remoteCharlie (Name "charlie")]
+        (postQualifiedMembers' g alice (remoteBob :| []) convId)
```

Comment:
Do I read this code correctly as "dear Mock Federator, please give back a profile for Charlie when asking for the userId of bob" ? If so, this is an interesting test case that is good to cover but implausible to happen in well-behaved backends. Could you also extend this test to include the (likely more common) case of the domain being correct but that user not existing (should give 404 in the client-server API, not sure what the RPC will give) or that user having been deleted (deleted flag in the profile being set to true)?

Reply:
I think what I meant to do here is add both Bob and Charlie to the conversation, where the remote only knows about Charlie. But I forgot to include Charlie in the request.

I will add the other tests you suggested, though, they do sound like good scenarios to check.